astropy/visualization/lupton_rgb.py

```diff
@@ -0,0 +1,469 @@
+
+        zscale_limits = ZScaleInterval().get_limits(image)
+        zscale = LinearMapping(*zscale_limits, image=image)
+        data_range = zscale.maximum - zscale.minimum[0]  # zscale.minimum is always a triple
+        minimum = zscale.minimum
+
+        for i, level in enumerate(pedestal):
+            minimum[i] += level
+
+        AsinhMapping.__init__(self, minimum, data_range, Q)
+        self._image = image
+
+
+def make_lupton_rgb(image_r, image_g=None, image_b=None, minimum=0, data_range=5, Q=8,
+                    saturated_border_width=0, saturated_pixel_value=None,
+                    x_size=None, y_size=None, rescale=None,
+                    filename=None):
+    """
+    Return a Red/Green/Blue color image from up to 3 images using an asinh stretch.
+    The input images can be int or float, and in any range or bit-depth.
+
+    For a more detailed look at the use of this method, see the document
+    :ref:`astropy-visualization-rgb`.
+
+    Parameters
+    ----------
+
+    image_r : `~numpy.ndarray`
+        Image to map to red (if None, use the image passed to the constructor).
+    image_g : `~numpy.ndarray`
+        Image to map to green (if None, use image_r).
+    image_b : `~numpy.ndarray`
+        Image to map to blue (if None, use image_r).
+    minimum : float
+        Intensity that should be mapped to black (a scalar or array for R, G, B).
+    data_range : float
+        minimum+data_range defines the white level of the image.
+    Q : float
+        The asinh softening parameter.
+    saturated_border_width : int
+        If saturated_border_width is non-zero, replace saturated pixels with saturated_pixel_value.
+    saturated_pixel_value : float
+        Value to replace saturated pixels with.
+    x_size : int
+        Desired width of RGB image (or None).  If y_size is None, preserve aspect ratio.
+    y_size : int
+        Desired height of RGB image (or None).
+    rescale : float
+        Make size of output image rescale*size of the input image.
+        Cannot be specified if x_size or y_size are given.
+    filename: str
+        Write the resulting RGB image to a file (file type determined frome extension).
+
+    Returns
+    -------
+    rgb : `~numpy.ndarray`
+        RGB (integer, 8-bits per channel) color image as an NxNx3 numpy array.
+    """
+    image_r = image_r
+    if image_g is None:
+        image_g = image_r
+    if image_b is None:
+        image_b = image_r
+
+    if saturated_border_width:
+        if saturated_pixel_value is None:
+            raise ValueError("saturated_pixel_value must be set if "
+                             "saturated_border_width is set.")
+        msg = "Cannot do this until we extract replaceSaturatedPixels out of afw/display/saturated.cc"
+        raise NotImplementedError(msg)
+        # replaceSaturatedPixels(image_r, image_g, image_b, saturated_border_width, saturated_pixel_value)
+
+    asinhMap = AsinhMapping(minimum, data_range, Q)
```

Comment:
I didn't look yet at the "Mapping" classes, but it would be good to be able to choose which mapping to use here, maybe with a string parameter ? Otherwise it means that, if I want to use the zscale mapping, I have to use the non-public API (`from astropy.visualization.lupton_rgb import AsinhZScaleMapping ; AsinhZScaleMapping.make_rgb_image(...)`).